autoload/neomake/makers/clippy.vim

```diff
@@ -0,0 +1,29 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#clippy#clippy() abort
+    let errorfmt = '%Eerror[E%n]: %m,'.
+                 \ '%Eerror: %m,'.
+                 \ '%Wwarning: %m,'.
+                 \ '%Inote: %m,'.
+                 \ '%-Z\ %#-->\ %f:%l:%c,'.
+                 \ '%I\ %#\= %t%*[^:]: %m,'.
+                 \ '%I\ %#|\ %#%\\^%\\+ %m,'.
+                 \ '%-G%s,'
+
+    " When rustup and a nightly toolchain is installed, that is used.
+    " Otherwise, the default cargo exectuable is used. If this is not part
+    " of a nightly rust, this will fail.
+    if executable('rustup') && system('rustup show | grep nightly | wc -l') >= 1
```

Comment:
As for matching this, we should use at least `grep -q '^nightly-'` then, and no `wc -l`.
You get the error code of `system` in v:shell_error` btw.